woocommerce/woocommerce-ios
# [Backend Receipts] Infer best printing paper size from web view page size

Closes: #11870 

## Description
This PR attempts to address the problem of wasted paper on backend receipts based on printer configuration settings. 

During testing we've seen that the default parameters we get from the printer might not be the best for receipt printing, and there's an excess of paper being used with the default settings.

## Changes
In order to minimize this problem we intercept and modify the `UIPrintInteractionControllerDelegate.printInteractionController(_:choosePaper:)`  delegate method to, rather than using the default paper size given by the system, return a paper size inferred from the webview's page size that holds the receipt.

If we cannot infer this size, we will just return the default paper provided by the system.

## Testing instructions
1. On an eligible store for backend receipts (using WooCommerce `8.6.0-dev` or `8.7.0-dev` versions. You can use https://indiemelon.mystagingwebsite.com/ if needed)
2. Go to Orders > Create a new order > Collect payment with any method
3. Tap on `See receipt` (pull to refresh or re-navigate to the order if needed)
4. Tap print and print the receipt on a physical printer

Ideally we should be able to infer a paper size based on the web view where the receipt has been rendered, which will return less wasted paper. If we cannot infer this, we'll see the _"Unable to retrieve inferred paper size from the web view. Using default paper provided by the system."_ message in the console.

It can also be helpful to add some debugging print statements to the `printInteractionController(_ printInteractionController: UIPrintInteractionController, choosePaper paperList: [UIPrintPaper]` delegate method to see the differences in estimated paper size by the system:
```
print("Estimated paper size: \(String(describing: paperSize))")
print("Best paper size: \(updatedPaper.paperSize)")
```
